trl/trainer/sft_trainer.py

```diff
@@ -233,7 +286,14 @@ def __init__(
             data_collator = DataCollatorWithFlattening()
 
         if data_collator is None:
-            data_collator = DataCollatorForLanguageModeling(tokenizer=processing_class, mlm=False)
+            # Get the pad token ID
+            pad_token_id = args.pad_token_id or processing_class.pad_token_id
+            if pad_token_id is None:
+                raise ValueError(
+                    "No `pad_token_id` was found. Either set `pad_token_id` in `SFTConfig`, or ensure that the "
+                    "tokenizer has a defined `pad_token`."
+                )
```

Comment:
Or, what about this?

```suggestion
            pad_token_id = args.pad_token_id or processing_class.pad_token_id or processing_class.eos_token_id
```